### DISCUSSION TITLE
encoder data array as input

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#725] [General]
Is it possible with your encoder library to feed  array of data as input and  output as  Another array .

Some thing like this 

return EncodedArray  SBCEncoderOut ( ArrayInput , LengthOfarray )

**Comment by pschatzmann** [comment]
I am afraid that I don't quite understand your question. 

I suggest that you go thru class documentation of the different layers of the involved classes.
For all of them you can write any size of bytes. The important thing is that you feed enough data within a given time so that you don't run out of audio data.

- [EncodedAudioStream](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_encoded_audio_stream.html) that is just a small layer on top of
- [Decoder](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_audio_decoder.html) that is forwarding the calls to
- The specific Codec Implementation

I suggest that you also look at the different Stream implementations that you can use to provide the result: e.g.
- CallbackStream
- QueuStream
- RingBufferStream
